txnsync/profiler_test.go

```diff
@@ -0,0 +1,276 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package txnsync
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/logging/telemetryspec"
+	"github.com/algorand/go-algorand/util/timers"
+)
+
+/*
+	Create a logger that hooks the "Metrics" function to signal that we have
+	indeed sent some metrics
+*/
+
+type MetricsLogger struct {
+	Logger
+	sentLogger *bool
+}
+
+func makeMetricsLogger(sentLogger *bool) MetricsLogger {
+	return MetricsLogger{
+		sentLogger: sentLogger,
+	}
+}
+
+func (n MetricsLogger) Metrics(category telemetryspec.Category, metrics telemetryspec.MetricDetails, details interface{}) {
+	*n.sentLogger = true
+}
+
+/*
```

Comment:
// instead of /* */
Also, it is a good practice to start the comment with the name of the function i.e. TestPrune
This is not required here, but golint looks for this to make sure exported functions have comments.